.github/workflows/test.yml

```
@@ -0,0 +1,453 @@
+        - mongodb: '4.2'
+          ruby: ruby-2.6
+          topology:
+          task: test
+          driver: current
+          rails: '5.2'
+          i18n:
+          gemfile: gemfiles/rails-5.2.gemfile
+          experimental: false
+        - mongodb: '4.2'
+          ruby: ruby-2.7
+          topology:
+          task: test
+          driver: current
+          rails: '5.2'
+          i18n:
+          gemfile: gemfiles/rails-5.2.gemfile
+          experimental: false
+        - mongodb: '4.4'
+          ruby: ruby-2.6
+          topology:
+          task: test
+          driver: current
+          rails: '6.0'
+          i18n:
+          gemfile: gemfiles/rails-6.0.gemfile
+          experimental: false
+        - mongodb: '4.4'
+          ruby: ruby-2.7
+          topology:
+          task: test
+          driver: current
+          rails: '6.0'
+          i18n:
+          gemfile: gemfiles/rails-6.0.gemfile
+          experimental: false
+        - mongodb: '4.4'
+          ruby: ruby-3.0
+          topology:
+          task: test
+          driver: current
+          rails: '6.0'
+          i18n:
+          gemfile: gemfiles/rails-6.0.gemfile
+          experimental: false
+        - mongodb: '4.2'
+          ruby: ruby-2.5
+          topology:
+          task: test
+          driver: current
+          rails:
+          i18n: '1.0'
+          gemfile: gemfiles/i18n-1.0.gemfile
+          experimental: false
+        - mongodb: '4.2'
+          ruby: ruby-2.6
+          topology:
+          task: test
+          driver: current
+          rails:
+          i18n: '1.0'
+          gemfile: gemfiles/i18n-1.0.gemfile
+          experimental: false
+        - mongodb: '4.2'
+          ruby: ruby-2.6
+          topology:
+          task: test
+          driver: current
+          rails:
+          i18n: current
+          gemfile: Gemfile
+          experimental: false
+        - mongodb: '4.4'
+          ruby: ruby-2.7
+          topology:
+          task: test
+          driver: current
+          rails: '5.2'
+          i18n:
+          gemfile: gemfiles/rails-5.2.gemfile
+          experimental: false
+    steps:
+    - name: repo checkout
+      uses: actions/checkout@v2
+    - name: start mongodb
+      uses: supercharge/mongodb-github-action@1.6.0
+      with:
+        mongodb-version: "${{matrix.mongodb}}"
+        mongodb-replica-set: rs0
+    - name: load ruby
+      uses: ruby/setup-ruby@v1
+      env:
+        BUNDLE_GEMFILE: "${{matrix.gemfile}}"
+      with:
+        ruby-version: "${{matrix.ruby}}"
+        bundler: 2
+    - name: bundle
+      run: |-
+        bundle install --jobs 4 --retry 3
+        git submodule init
+        git submodule update
+      env:
+        BUNDLE_GEMFILE: "${{matrix.gemfile}}"
+    - name: test
+      timeout-minutes: 60
+      continue-on-error: "${{matrix.experimental}}"
+      run: bundle exec rake spec
+      env:
+        BUNDLE_GEMFILE: "${{matrix.gemfile}}"
```

Not sure how the connection string is passed to the driver, but in PHP we use the `MONGODB_URI` env variable that needs to be set to the output of the `start mongodb` step earlier:
```suggestion
        BUNDLE_GEMFILE: "${{matrix.gemfile}}"
        MONGODB_URI: ${{ steps.setup-mongodb.outputs.cluster-uri }}
```